### PR TITLE
Remove minimal implementations for security analyzers

### PR DESCRIPTION
**Pull request by Evangelink**
Contributes to #3276, see #3279

### CODE REVIEW

File: src/Test.Utilities/AdditionalMetadataReferences.cs [hunk 1]
```diff
@@ -29,6 +30,18 @@ public static class AdditionalMetadataReferences
         public static ReferenceAssemblies DefaultWithNewtonsoftJson { get; } = Default
             .AddPackages(ImmutableArray.Create(new PackageIdentity("Newtonsoft.Json", "10.0.1")));
 
+        public static ReferenceAssemblies DefaultWithWinForms { get; } = Default
+            .AddAssemblies(ImmutableArray.Create("System.Windows.Forms"));
+
+        public static ReferenceAssemblies DefaultWithWinHttpHandler { get; } = ReferenceAssemblies.NetStandard.NetStandard20
```
Comment (Evangelink) [inline src/Test.Utilities/AdditionalMetadataReferences.cs]: NetFramework assemblies were causing an issue about `HttpClient` being imported twice.

File: src/Test.Utilities/AdditionalMetadataReferences.cs [hunk 2]
```diff
@@ -29,6 +30,18 @@ public static class AdditionalMetadataReferences
...
+            .AddPackages(ImmutableArray.Create(new PackageIdentity("System.Net.Http.WinHttpHandler", "4.7.0")));
+
+        public static ReferenceAssemblies DefaultWithAspNetCoreMvc { get; } = Default
+            .AddPackages(ImmutableArray.Create(
+                new PackageIdentity("Microsoft.AspNetCore", "1.1.7"),
+                new PackageIdentity("Microsoft.AspNetCore.Mvc", "1.1.8"),
+                new PackageIdentity("Microsoft.AspNetCore.Http", "1.1.2")));
```
Comment (Evangelink) [inline src/Test.Utilities/AdditionalMetadataReferences.cs]: More recent versions of the framework are causing some tests to fail.

**Comment by Evangelink** [comment]
Oh yeah I forgot to mention that I cannot find the package containing the `CurlHandler`

File: src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Security/DoNotDisableHttpClientCRLCheckTests.cs [hunk 3]
```diff
@@ -22,10 +21,8 @@ private async Task VerifyCSharpWithDependenciesAsync(string source, params Diagn
         {
             var csharpTest = new VerifyCS.Test
             {
-                TestState =
-                {
-                    Sources = { source, SystemNetHttpApis.CSharp },
```
Comment (sharwell) [inline src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Security/DoNotDisableHttpClientCRLCheckTests.cs]: 💭 System.Net.Http is included by default in netstandard2.0. Are you seeing errors if we just let it use default?

**Comment by mavasani** [comment]
Tagging @dotpaul for review.

File: src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Security/DoNotDisableHttpClientCRLCheckTests.cs (hunk 3 above)
Comment (Evangelink) [reply src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Security/DoNotDisableHttpClientCRLCheckTests.cs]: Yes there were errors about `CurlHandler`, `WinHttpHandler` and a third one that I don't remember.

**Comment by dotpaul** [comment]
> Oh yeah I forgot to mention that I cannot find the package containing the `CurlHandler`.

Hrm, looks like CurlHandler was internal (https://github.com/dotnet/corefx/blob/d3911035f2ba3eb5c44310342cc1d654e42aa316/src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.SslProvider.Linux.cs#L21) and I can't find it in the new dotnet/runtime repo. I'd be fine with removing the CurlHandler parts from the rule itself. Or I can do it you'd prefer not to.

File: src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Security/DoNotDisableHttpClientCRLCheckTests.cs (hunk 3 above)
Comment (Evangelink) [reply src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Security/DoNotDisableHttpClientCRLCheckTests.cs]: I confirm that there are errors when I don't use the custom ref assemblies. I have even tried to force `ReferenceAssemblies.NetStandard.Netstarnd2.0`

**Comment by dotpaul** [comment]
@dotnet-bot retest please

**Pull request merged**